src/internal/packager2/load.go

```diff
@@ -0,0 +1,212 @@
+	PublicKeyPath           string
+	SkipSignatureValidation bool
+	Filter                  filters.ComponentFilterStrategy
+}
+
+// LoadPackage optionally fetches and loads the package from the given source.
+func LoadPackage(ctx context.Context, opt LoadOptions) (*layout.PackagePaths, error) {
+	srcType, err := identifySource(opt.Source)
+	if err != nil {
+		return nil, err
+	}
+
+	tarDir, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tarDir)
+	tarPath := filepath.Join(tarDir, "packager.tar")
+	isPartial := false
+	switch srcType {
+	case "oci":
+		isPartial, err = pullOCI(ctx, opt.Source, tarPath, opt.Shasum, opt.Filter)
+		if err != nil {
+			return nil, err
+		}
+	case "http", "https":
+		err = pullHTTP(ctx, opt.Source, tarPath, opt.Shasum)
+		if err != nil {
+			return nil, err
+		}
+	case "split":
+		err = assembleSplitTar(opt.Source, tarPath)
+		if err != nil {
+			return nil, err
+		}
+	case "tarball":
+		tarPath = opt.Source
+	default:
+		return nil, fmt.Errorf("unknown source type: %s", opt.Source)
+	}
+	if srcType != "oci" && opt.Shasum != "" {
+		err := helpers.SHAsMatch(tarPath, opt.Shasum)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Extract the package
+	packageDir, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return nil, err
+	}
+	pathsExtracted := []string{}
+	err = archiver.Walk(tarPath, func(f archiver.File) error {
+		if f.IsDir() {
+			return nil
+		}
+		header, ok := f.Header.(*tar.Header)
+		if !ok {
+			return fmt.Errorf("expected header to be *tar.Header but was %T", f.Header)
+		}
+		// If path has nested directories we want to create them.
+		dir := filepath.Dir(header.Name)
+		if dir != "." {
+			err := os.MkdirAll(filepath.Join(packageDir, dir), helpers.ReadExecuteAllWriteUser)
+			if err != nil {
+				return err
+			}
+		}
+		dst, err := os.Create(filepath.Join(packageDir, header.Name))
+		if err != nil {
+			return err
+		}
+		defer dst.Close()
+		_, err = io.Copy(dst, f)
+		if err != nil {
+			return err
+		}
+		pathsExtracted = append(pathsExtracted, header.Name)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Load the package paths
+	pkgPaths := layout.New(packageDir)
+	pkgPaths.SetFromPaths(pathsExtracted)
+	pkg, _, err := pkgPaths.ReadZarfYAML()
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Filter is not persistently applied.
+	pkg.Components, err = opt.Filter.Apply(pkg)
+	if err != nil {
+		return nil, err
+	}
+	if err := pkgPaths.MigrateLegacy(); err != nil {
+		return nil, err
+	}
+	if !pkgPaths.IsLegacyLayout() {
+		if err := sources.ValidatePackageIntegrity(pkgPaths, pkg.Metadata.AggregateChecksum, isPartial); err != nil {
+			return nil, err
+		}
+		if opt.SkipSignatureValidation {
+			if err := sources.ValidatePackageSignature(ctx, pkgPaths, opt.PublicKeyPath); err != nil {
+				return nil, err
+			}
+		}
+	}
+	for _, component := range pkg.Components {
```

Comment:
Will this return errors now without `unarchiveAll`

Reply:
The bool `unarchiveAll` is only false in the publish logic. I done really understand why this is. Personally I want to get away from all the different states that a package path can exist in by removing options.